src/common.ts

```diff
@@ -96,14 +94,26 @@ export const nullable = <T>(processor: ValueProcessor<T>): ValueProcessor<T | nu
   },
 });
 
-export const nullOr = <T, O>(factory: ValueProcessorFactory<T, O>): ValueProcessorFactory<T | null, O> => (options) => nullable(factory(options));
+export const nullOr = <T>(processor: ValueProcessor<T>): ValueProcessor<T | null> => nullable(processor);
+
+export const nullableAs = <T>(processor: ValueProcessor<T>, options?: WithDefault<T | null>): ValueProcessor<Exclude<T, undefined> | null> => ({
+  process: (value: unknown): Result<Exclude<T, undefined> | null> => {
+    if (value === null) return { value: null };
+
+    const result = processor.process(value);
+    if (options?.default !== undefined && !isIssue(result) && result.value === undefined) return { value: options.default as Exclude<T, undefined> | null };
```

Comment:
intead of `return` you can overwrite `result.value`

Reply:
the type of result.value won't accept the type of the default